src/WebJobs.Script.WebHost/Diagnostics/DiagnosticEventTableStorageRepository.cs

```diff
@@ -41,43 +41,30 @@ public class DiagnosticEventTableStorageRepository : IDiagnosticEventRepository,
         private string _tableName;
         private bool _isEnabled = true;
 
-        internal DiagnosticEventTableStorageRepository(IConfiguration configuration, IHostIdProvider hostIdProvider, IEnvironment environment, IScriptHostManager scriptHostManager,
-            ILogger<DiagnosticEventTableStorageRepository> logger, int logFlushInterval)
+        internal DiagnosticEventTableStorageRepository(IHostIdProvider hostIdProvider, IEnvironment environment, IScriptHostManager scriptHostManager,
+            IAzureTableStorageProvider azureTableStorageProvider, ILogger<DiagnosticEventTableStorageRepository> logger, int logFlushInterval)
         {
-            _configuration = configuration;
             _hostIdProvider = hostIdProvider;
             _environment = environment;
             _serviceProvider = scriptHostManager as IServiceProvider;
             _logger = logger;
             _flushLogsTimer = new Timer(OnFlushLogs, null, logFlushInterval, logFlushInterval);
+            _azureTableStorageProvider = azureTableStorageProvider;
         }
 
-        public DiagnosticEventTableStorageRepository(IConfiguration configuration, IHostIdProvider hostIdProvider, IEnvironment environment, IScriptHostManager scriptHost,
-            ILogger<DiagnosticEventTableStorageRepository> logger)
-            : this(configuration, hostIdProvider, environment, scriptHost, logger, LogFlushInterval) { }
+        public DiagnosticEventTableStorageRepository(IHostIdProvider hostIdProvider, IEnvironment environment, IScriptHostManager scriptHost,
+            IAzureTableStorageProvider azureTableStorageProvider, ILogger<DiagnosticEventTableStorageRepository> logger)
+            : this(hostIdProvider, environment, scriptHost, azureTableStorageProvider, logger, LogFlushInterval) { }
 
         internal TableServiceClient TableClient
         {
             get
             {
-                if (!_environment.IsPlaceholderModeEnabled() && _tableClient == null)
+                if (!_environment.IsPlaceholderModeEnabled() && _tableClient == null && !_azureTableStorageProvider.TryCreateHostingTableServiceClient(out _tableClient))
                 {
-                    string storageConnectionString = _configuration.GetWebJobsConnectionString(ConnectionStringNames.Storage);
-
-                    try
-                    {
-                        _tableClient = new TableServiceClient(storageConnectionString);
-
-                        // The TableServiceClient only verifies the format of the connection string.
-                        // To ensure the storage account exists and supports Table storage, validate the connection string by retrieving the properties of the table service.
-                        _ = _tableClient.GetProperties();
```

Comment:
Removed this check since it could disable Diagnostic events for some scenarios where we might not want to:
- The permissions assigned to the Identity don't allow to get the properties but allow to write diagnostic events
- Transient network issue on `TableServiceClient` creation that could be auto-resolved so we could write events on that instance in the future
- Permissions: missing permissions on `TableServiceClient` creation that are either solved by the customer later on or take time to be propagated, which would fix writing diagnostic events without a restart.

Reply:
Does this mean we could see errors somewhere else later? I assume this check was added for that reason at some point?

Reply:
We added that check for the current release in #10395, to disable the repository if the storage account connection string was not present. The main issue was due to the fact that when you were using Identity based connections, we would have tons of logs about not having the connection string to be used. That situation is still resolved because with this change we support identity based connections and also disabled the repo if we fail to create the client.

The `GetProperties()` call was also added because that way we could disable the repository on a situation that we knew there was issues connecting to Table Storage, right on client initialization. However while adding support for MI, I've found and tested some cases where would be good to let it fail as they would be recoverable errors. Not all the cases ofc, for example an invalid account name would require a config change and a restart, but that one would also other hard-failures host-wise....

After weighing it up, I decided to remove that check.

Reply:
Should we check for a `403` response and disable the repository when we try and write?

Reply:
We could indeed, but we might still have some scenarios where the situation could be resolved (such role/permissions propagation) without requiring a host restart and on those scenarios we would be disabling prematurely. There are some SKUs where the same host can be running without being restarted for longer periods so there it could impact more.